doc/fwutil/fwutil.md

```diff
@@ -462,6 +498,278 @@ Aborted!
+6. -z|--fw-image - firmware package downloaded during run time (this is an exclusive option from --image)
+
+**Note:** the default option is _--image=current_, _--boot=any_,and  _--action=none_ 
+
+##### 2.2.2.4.3 Component Firmware Update Script requirement
+
+##### 2.2.2.4.3.1 Overview
+
+The purpose of the <component>_fw_upgrade commands group is to provide an interface for:
+1. The specific platform component related information query (version, description, etc.)
+2. Available component FW update related information query (fw, version, status, etc.)
+2. Update the component firmware based on the following boot-action
+
+##### 2.2.2.4.3.1.1 Description
+
+**The following command displays FW utility version:**
+<component>
+```bash
+root@sonic:~# ssd_fw_upgrade show version
+ssd_fw_upgrade version 1.0.0.0
+```
+
+**The following command displays platform component name and FW version:**
+```bash
+root@sonic:~# ssd_fw_upgrade show status
+Component   Version             Description
+---------  ------------------  ------------
+SSD        5                    <vendor>/<model>
+```
+
+**The following command displays available FW updates:**
+```bash
+root@sonic:~# ssd_fw_update show updates ./ssd_fw.bin
+Component   Firmware               Version (current/available)              Status              Required Boot Action
+---------   ---------------------  ---------------------------------------  ------------------  -----------------------
+SSD         <image_path>/ssd.bin   4 / 5                                    update is required  Immediate Cold/Fast reboot
+```
+
+**The following command performs the component FW update:**
+```bash
+root@sonic:~# ssd_fw_update --boot=immediate_cold ./ssd_fw.bin
+Warning: <firmware_update_notification>
+...
+FW update in progress ...
+...
+```
+
+**Supported options:**
+1. -b|--boot - following boot option after the upgrade
+
+**Note:** the default option is _--boot=any_
+
+##### 2.2.2.4.4 Auto-update Use Cases
+
+##### 2.2.2.4.4.1 FW-UPDATE-PLUGIN for reboot scripts - reboot/fast-reboot/warm-reboot
+During any reboot, device can update with any available firmware image if the reboot type is applicable for the firmware  update.
+```bash
+root@sonic:~# fwutil auto-update --yes --image=next --boot=immediate_fast
+```
+Example : Reboot script changes to support automatic firmware update during a reboot
+
+    ```
+    ...
+    from sonic_installer.bootloader import get_bootloader
+    ...
+    PLATFORM_FW_UPDATE="/usr/bin/fwutil"
+    ...
+    bootloader = get_bootloader()
+    curimage = bootloader.get_current_image()
+    nextimage = bootloader.get_next_image()
+    If [[ nextimage == curimage ]]; then
+        NEXT_BOOT_IMAGE=current
+    else
+        NEXT_BOOT_IMAGE=next
+    fi
+    ...
+    if [[ -x ${PLATFORM_FW_UPDATE} ]]; then
+        debug "updating platform fw for ${REBOOT_TYPE}"
+        ${PLATFORM_FW_UPDATE} auto-update --yes --image={NEXT_BOOT_IMAGE} --boot=${REBOOT_TYPE}
```

Comment:
Hi Sujin,

In what scenarios the fwutil command will include a reboot/powercycle? only in cases of immediate_(fast/warm/cold/powercycle)?

Also in few cases the upgrade may require unmount of filesystems and better to include them in the flow of reboot after unmounting filesystems. In that case fwutil can do just the placing of files in correct location and the actual upgrade will be done only as part of the reboot script. In those scenarios if this command just places the file, in what ways the actual upgrade can be included in these scripts?

Reply:
Do you know the answer about your first question? I also have this question.
####In what scenarios the fwutil command will include a reboot/powercycle? only in cases of immediate_(fast/warm/cold/powercycle)?####